internal/pkg/agent/storage/encrypted_disk_storage_windows_linux_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux || windows
+// +build linux windows
+
+package storage
+
+import (
+	"bytes"
+	"errors"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/secret"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/vault"
+	"github.com/google/go-cmp/cmp"
+)
+
+const (
+	testConfigFile = "someconfig.enc"
+	vaultDir       = "vault"
+)
+
+func TestEncryptedDiskStorageWindowsLinuxLoad(t *testing.T) {
+
+	// Disable root permissions check
+	vault.DisableRootCheck()
+
+	dir := t.TempDir()
+
+	fp := filepath.Join(dir, testConfigFile)
+	s := NewEncryptedDiskStore(fp, WithVaultPath(dir))
+
+	// Test that the file loads and doesn't create vault
+	r, err := s.Load()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r.Close()
+
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Expect empty content from the reader
+	diff := cmp.Diff(0, len(b))
+	if diff != "" {
+		t.Error(diff)
+	}
+
+	// Expect no vault directory was not created
```

Comment:
Typo?
This double negation seems odd to me...
```suggestion
	// Expect no vault directory was created
```

Reply:
fixed the typo